app/controllers/v1/sessions_controller.rb

```diff
@@ -385,15 +385,26 @@ def after_login_actions
     def log_persisted_session_and_warnings
       obscure_token = Session.obscure_token(@session_object.token)
       Rails.logger.info("Logged in user with id #{@session_object&.uuid}, token #{obscure_token}")
-      # We want to log when SSNs do not match between MVI and SAML Identity. And might take future
-      # action if this appears to be happening frequently.
-      if current_user.ssn_mismatch?
-        additional_context = StringHelpers.heuristics(current_user.identity.ssn, current_user.ssn_mpi)
-        log_message_to_sentry(
-          'SessionsController version:v1 message:SSN from MPI Lookup does not match UserIdentity cache',
-          :warn,
-          identity_compared_with_mpi: additional_context
-        )
+      id_mismatch_validations
+    end
+
+    def id_mismatch_validations
+      return unless current_user.loa3?
+
+      check_id_mismatch(@current_user.ssn, @current_user.ssn_mpi, :identity_mpi_conflict_ssn)
+      check_id_mismatch(@current_user.edipi, @current_user.edipi_mpi, :identity_mpi_conflict_edipi)
+      check_id_mismatch(@current_user.icn, @current_user.mpi_icn, :identity_mpi_conflict_icn)
+      check_id_mismatch(@current_user.mhv_correlation_id, @current_user.mpi_mhv_correlation_id,
+                        :identity_mpi_conflict_mhv_correlation_id)
+    end
+
+    def check_id_mismatch(identity_value, mpi_value, id_mismatch_type)
+      return unless identity_value.present? && mpi_value.present?
+
+      if identity_value != mpi_value
+        error_message = SAML::UserAttributeError::ERRORS[id_mismatch_type][:message]
+        error_data = { identity_value: identity_value, mpi_value: mpi_value, icn: current_user.icn }
```

Comment:
Do we think it's an issue that we could potentially log SSN here? We might want to add an additional param on `check_id_mismatch` function that's something like 'obscure_data' which prevents the actual values from being logged (only enabled on the ssh values)